TLM/TMPE.API/Traffic/Data/SpeedValue.cs

```diff
@@ -0,0 +1,104 @@
+namespace TrafficManager.API.Traffic.Data {
+    using System;
+    using UnityEngine;
+
+    /// <summary>
+    /// Represents a speed value expressed in km/hour.
+    /// </summary>
+    [Serializable]
+    public struct KmphValue {
+        public KmphValue(ushort kmph) {
+            Kmph = kmph;
+        }
+
+        public override string ToString() {
+            return $"{Kmph:0.0} km/h";
+        }
+
+        public ushort Kmph { get; set; }
+    }
+
+    /// <summary>
+    /// Represents a speed value expressed in miles/hour.
+    /// </summary>
+    [Serializable]
+    public struct MphValue {
+        public MphValue(ushort mph) {
+            Mph = mph;
+        }
+
+        public override string ToString() {
+            return $"{Mph} MPH";
+        }
+
+        public ushort Mph { get; set; }
+    }
+
+    /// <summary>
+    /// Represents a speed value expressed in game units where 1f = 50 km/h or 32 MPH.
+    /// </summary>
+    [Serializable]
+    public struct SpeedValue {
```

Comment:
This struct would benefit from some commenting of the fields/properties/methods.

Reply:
Some XML documentation has been added